google/auth/compute_engine/credentials.py

```diff
@@ -120,7 +115,7 @@ def requires_scopes(self):
 _DEFAULT_TOKEN_URI = 'https://www.googleapis.com/oauth2/v4/token'
 
 
-class IDTokenCredentials(credentials.Credentials, credentials.Signing):
+class IDTokenCredentials(credentials.Credentials):
```

Comment:
Removing `Signing` breaks the existing interface of this class. Why can't these credentials continue to sign using the IAM API?

Reply:
Yeah, but the capability shouldn't be there anyway, IMO.  GCE itself doens' thave the capability to sign: it can get `access_tokens` and even an `id_token` but not sign arbitrary blobs by itself.  IMO the computeCredential object should just reflect its baseline capabilities the underlying provider gives here anyway (in this case, the provider is the metadata service).  

The `credentials.Signing` capability is baked into the existing class as really just a side effect  too interms of gettig an IDtoken:  if you were using the IAM api, you could 'just get' an id_token [directly](https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/generateIdToken)  without signing

Anyway, her'es the biggest problem i see,  that in the existing implementation, the singer and sign_bytes was already _exposed_  directly [here](https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/compute_engine/credentials.py#L229)

```python
    def signer(self):
        return self._signer

    def sign_bytes(self, message):
        return self._signer.sign(message)
```

I don't think people are using that capability to direct sign via IAM but it is there.

---

What the existing implementation is doing is basically allowing the GCE to "impersonate itself" and use the IAM API to get an id token and along the way also allow blob signing.   I would say if someone wants to sigin via the IAM api, the `impersonated_credential` object could do that (i'll file a separate PR for something like [this](https://gist.github.com/salrashid123/5caaa6178e786bfb48515a2cfb7631af#file-impersonated_credentials-py-L242) later

Reply:
> don't think people are using that capability to direct sign via IAM but it is there.

This isn't a good enough reason for us to break a 1.0 interface. this will be a breaking change to users and it will cause issues.

Reply:
no, thats not the reason (its just a statement).  the reasons are outlined above as well as the ability to sign isn't privilege a GCE instance has.  

per offline, we'll can keep the singer capability as well as using the IAM API to do that bit.  Getting the ID token will be through the metadta sever per the most recent commit.